sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/SocketClient.java

```diff
@@ -0,0 +1,148 @@
+        int port = httpRequest.getUrl().getPort();
+        if (httpRequest.getUrl().getProtocol().equals("https")) {
+            SSLSocketFactory sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            try (SSLSocket socket = (SSLSocket) sslSocketFactory.createSocket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        } else if (httpRequest.getUrl().getProtocol().equals("http")) {
+            try (Socket socket = new Socket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        }
+        throw new ProtocolException("Only HTTP and HTTPS are supported by this client.");
+    }
+
+    /**
+     * Gets a String representation of the request and writes it to the output stream,
+     * then calls buildResponse to get an instance of HttpUrlConnectionResponse from the
+     * input stream
+     *
+     * @param httpRequest {@link com.azure.core.http.HttpRequest} instance
+     * @param socket {@link java.net.Socket} instance
+     * @return an instance of HttpUrlConnectionResponse
+     */
+    private static HttpUrlConnectionResponse doInputOutput(HttpRequest httpRequest, Socket socket) throws IOException {
+        httpRequest.setHeader(HttpHeaderName.HOST, httpRequest.getUrl().getHost());
+        String request = buildPatchRequest(httpRequest);
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
+             OutputStreamWriter out = new OutputStreamWriter(socket.getOutputStream())) {
+
+            out.write(request);
+            out.flush();
+
+            HttpUrlConnectionResponse response = buildResponse(httpRequest, in);
+
+            String redirectLocation = response.getHeaders().stream()
+                .filter(h -> h.getName().equals("Location"))
+                .map(HttpHeader::getValue)
+                .findFirst()
+                .orElse(null);
+
+            if (redirectLocation != null) {
+                httpRequest.setUrl(redirectLocation);
+                return sendPatchRequest(httpRequest);
+            }
+            return response;
+        }
+    }
+
+    /**
+     * Converts an instance of HttpRequest to a String representation for sending
+     * over the output stream
+     *
+     * @param httpRequest {@link com.azure.core.http.HttpRequest} instance
+     * @return the String representation of the HttpRequest
+     */
+    private static String buildPatchRequest(HttpRequest httpRequest) {
+        final StringBuilder request = new StringBuilder();
+
+        request.append("PATCH")
+            .append(" ")
+            .append(httpRequest.getUrl().getPath())
+            .append(HTTP_VERSION)
+            .append("\r\n");
+
+        if (httpRequest.getHeaders().getSize() > 0) {
+            for (HttpHeader headerLine : httpRequest.getHeaders()) {
+                request.append(headerLine.getName())
```

Comment:
Should multi-value headers be sent as

```
headername: headervalue1,headervalue2
```

or

```
headername: headervalue1
headername: headervalue2
```
?

Right, now this is doing the first case.